src/otel_telemetry.erl

```diff
@@ -54,41 +93,31 @@ tracer_id_for_events(Prefix, [Suffix | _], AllEvents) ->
 
 handle_event(_Event,
              #{system_time := StartTime},
-             _Metadata,
-             #{type := start, tracer := Tracer, span_name := Name}) ->
+             Metadata,
+             #{type := start, tracer_id := TracerId, span_name := Name}) ->
     StartOpts = #{start_time => StartTime},
-    _ = ot_tracer:start_span(Tracer, Name, StartOpts),
+    Tracer = opentelemetry:get_tracer(TracerId),
+    Ctx = otel_tracer:start_span(Tracer, Name, StartOpts),
+    _ = store_ctx(Ctx, TracerId, Metadata),
     ok;
 handle_event(_Event,
              #{duration := Duration},
-             _Metadata,
-             #{type := stop, tracer := Tracer}) ->
-    Ctx = ot_tracer:current_span_ctx(Tracer),
-    _ = ot_span:set_attribute(Tracer, Ctx, <<"duration">>, Duration),
-    _ = ot_tracer:end_span(Tracer, Ctx),
+             Metadata,
+             #{type := stop, tracer_id := TracerId}) ->
+    Ctx = pop_ctx(TracerId, Metadata),
+    otel_span:set_attribute(Ctx, <<"duration">>, Duration),
```

Comment:
I don't remember where I read this—I thought it was in the semantic convention but it isn't—but adding a suffix of what kind of time we're talking about a good practice? Like `duration_ms`

Reply:
That is a convention in Prometheus, but in Otel it's all assumed milliseconds.